src/main/java/nl/knaw/dans/filemigration/core/DataverseLoader.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2021 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.filemigration.core;
+
+import nl.knaw.dans.filemigration.api.ActualFile;
+import nl.knaw.dans.filemigration.db.ActualFileDAO;
+import nl.knaw.dans.lib.dataverse.DataverseClient;
+import nl.knaw.dans.lib.dataverse.model.dataset.DatasetVersion;
+import nl.knaw.dans.lib.dataverse.model.file.DataFile;
+import nl.knaw.dans.lib.dataverse.model.file.FileMeta;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class DataverseLoader {
+  private static final Logger log = LoggerFactory.getLogger(DataverseLoader.class);
+
+  private final ActualFileDAO actualFileDAO;
+  private final DataverseClient client;
+
+  public DataverseLoader(DataverseClient client, ActualFileDAO actualFileDAO) {
+    this.actualFileDAO = actualFileDAO;
+    this.client = client;
+  }
+
+  public void saveActual(ActualFile actual) {
+    log.debug(actual.toString());
+    actualFileDAO.create(actual);
+  }
+
+  public void loadFromDataset(String doi) {
+    log.info("Loading {}", doi);
+    List<DatasetVersion> versions;
+    try {
+      versions = client.dataset(doi).getAllVersions().getData();
```

Comment:
Stumbles on https://demo.archaeology.datastations.nl/api/datasets/export?exporter=dataverse_json&persistentId=doi%3A10.17026/dans-xtz-qa6j

with

    java.lang.RuntimeException: com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException: Unrecognized field "originalFileFormat" (class nl.knaw.dans.lib.dataverse.model.file.DataFile), not marked as ignorable (13 known properties: "rootDataFileId", "pidURL", "filename", "creationDate", "persistentId", "contentType", "previousDataFileId", "id", "description", "filesize", "checksum", "storageIdentifier", "embargo"])
     at [Source: (StringReader); line: 1, column: 29161] (through reference chain: nl.knaw.dans.lib.dataverse.model.DataverseEnvelope["data"]->java.util.ArrayList[0]->nl.knaw.dans.lib.dataverse.model.dataset.DatasetVersion["files"]->java.util.ArrayList[28]->nl.knaw.dans.lib.dataverse.model.file.FileMeta["dataFile"]->nl.knaw.dans.lib.dataverse.model.file.DataFile["originalFileFormat"])

No success in `DdVerifyFileMigrationApplication.initialize` with

    bootstrap.getObjectMapper().disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);